File: LignePanier.java
public class LignePanier {

    private Produit produit;
    private int quantite;

    public LignePanier(Produit produit, int quantite) {
        this.produit = produit;
        this.quantite = quantite;
    }

    public int getQuantite() {
        return quantite;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    public double getSousTotal() {
        return produit.getPrix() * quantite;
    }

    @Override
    public String toString() {
        return produit.getNom() + " | Prix: " + produit.getPrix() + " | Qté: " + quantite + " | Sous-total: " + getSousTotal();
    }
}

File: LignePanier.py
class LignePanier:

    # Constructeur
    def __init__(self, produit, quantite):
        self.produit = produit
        self.quantite = quantite

    def get_quantite(self):
        return self.quantite

    def set_quantite(self, quantite):
        self.quantite = quantite

    def get_sous_total(self):
        return self.produit.get_prix() * self.quantite

    def __str__(self):
        return (
            self.produit.get_nom()
            + " | Prix: "
            + str(self.produit.get_prix())
            + " | Qté: "
            + str(self.quantite)
            + " | Sous-total: "
            + str(self.get_sous_total())
        )
